### PR TITLE
adds passphrase support

### PR DESCRIPTION
**Pull request by constcrist**

### CODE REVIEW

**Review by poandrei — APPROVED** [review]

File: anonym-ip.go [hunk 1]
```diff
@@ -19,8 +20,46 @@ import (
...
+func NewCipher(key []byte) (cipher.Block, error) {
+	switch k := len(key); k {
+	default:
+		return nil, ErrKeySize
+	case BlockSize:
+		break
+	}
+	var a [BlockSize]byte
+	for i := range key {
+		a[i] = key[i]
```
Comment (poandrei) [inline anonym-ip.go:44]: minor: if you use range you could use both values:
```suggestion
	for i,v := range key {
		a[i] = v
```
you could also use copy:
```suggestion
	copy(a[:], key)
```
Not sure what would be faster (the copy version creates a tmp slice over the array).

File: anonym-ip_test.go [hunk 2]
```diff
@@ -0,0 +1,341 @@
...
+)
+
+func TestMain(m *testing.M) {
+	// initialize whatever needs to be initialized
+	os.Exit(m.Run())
+}
+
+func TestEncryptIP(t *testing.T) {
+	// set-up if needed
+	t.Run("IPv4, passphrase", func(t *testing.T) {
```
Comment (poandrei) [inline anonym-ip_test.go:17]: why not having separate function (like TestEncryptIPv4Passphrase)? Any advantage?

File: anonym-ip_test.go [hunk 3]
```diff
@@ -0,0 +1,341 @@
...
+		cases := []net.IP{
+			[]byte{1, 2, 3, 4},
+			[]byte{198, 41, 56, 22},
+			[]byte{22, 11, 33, 44},
+			[]byte{255, 0, 255, 241},
+		}
```
Comment (poandrei) [inline anonym-ip_test.go:23]: Possible addition: random IP & random key (e.g. generate 1000 random IP and test encrypt /decrypt with random key/password).

**Review by poandrei — APPROVED** [review]

File: anonym-ip_test.go (hunk 2 above)
Comment (constcrist) [reply anonym-ip_test.go:17]: grouping tests in tests suites. see:

https://golang.org/pkg/testing/
"Subtests and Sub-benchmarks"

you can run only sub-tests of a group of tests:
`go test -run EncryptIP/IPv4`

**Pull request merged**